vermouth/molecule.py

```diff
@@ -348,30 +348,48 @@ def atoms(self):
             node_attr = self.node[node]
             yield node, node_attr
 
-    def copy(self, as_view=False):
-        '''
+    def copy(self):
+        """
         Creates a copy of the molecule.
 
-        See Also
-        --------
-        :meth:`networkx.Graph.copy`
-        '''
-        copy = super().copy(as_view)
-        if not as_view:
-            copy = self.__class__(copy)
-        copy._force_field = self.force_field
-        copy.meta = self.meta.copy()
-        return copy
-
-    def subgraph(self, *args, **kwargs):
-        '''
+        Returns
+        -------
+        Molecule
+        """
+        return self.subgraph(self.nodes)
+
+    def subgraph(self, nodes):
+        """
         Creates a subgraph from the molecule.
 
-        See Also
-        --------
-        :meth:`networkx.Graph.subgraph`
-        '''
-        return self.__class__(super().subgraph(*args, **kwargs))
+
+        Returns
+        -------
+        Molecule
+        """
+        subgraph = self.__class__()
+        subgraph.meta = copy.copy(self.meta)
+        subgraph._force_field = self._force_field
+        subgraph.nrexcl = self.nrexcl
+
+        node_copies = [(node, copy.copy(self.nodes[node])) for node in nodes]
+        subgraph.add_nodes_from(node_copies)
+
+        nodes = set(nodes)
+
+        edges_to_add = [
+            (node, node2)
+            for node in nodes
+            for node2 in set(self[node]) & nodes
+        ]
+        subgraph.add_edges_from(edges_to_add)
```

Comment:
`subgraph.add_edges_from(self.edges_between(nodes, nodes))`? And this might loose edge data.